src/resource-manager.ts

```diff
@@ -1,44 +1,51 @@
 import axios from 'axios';
-
-export interface RobotResource {
-  icons: Record<string, string>; // Record<ModelName|FleetName, IconPath>
-}
+import { RobotResourceManager, RobotResource } from './resource-manager-robots';
+import { DispenserResourceManager, DispenserResource } from './resource-manager-dispensers';
 
 export interface ResourceConfigurationsType {
-  robots?: Record<string, RobotResource>; // Record<FleetName, RobotResource>
+  robots: Record<string, RobotResource>; // Record<FleetName, RobotResource>
```

Comment:
Instead of using `Partial` at most places, maybe we can have robots as optional and use `Required` in the few places (if there is any) that needs to assume that the properties are there.